cloud/azure/management/costs/mode/tagonresources.pm

```diff
@@ -0,0 +1,124 @@
+#
+# Copyright 2022 Centreon (http://www.centreon.com/)
+#
+# Centreon is a full-fledged industry-strength solution that meets
+# the needs in IT infrastructure and application monitoring for
+# service performance.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package cloud::azure::management::costs::mode::tagonresources;
+
+use base qw(centreon::plugins::templates::counter);
+
+use strict;
+use warnings;
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
+        'vms'               => { name => 'vms' },
+        'tag-name:s'        => { name => 'tag_name' },
+	'exclude-name:s'    => { name => 'exclude_name' }
+    });
+
+    return $self;
+}
+
+sub check_options {
+    my ($self, %options) = @_;
+    $self->SUPER::check_options(%options);
+
+    if (!defined($self->{option_results}->{tag_name}) || $self->{option_results}->{tag_name} eq '') {
+	$self->{output}->add_option_msg(short_msg => "Need to specify --tag-name option");
+	$self->{output}->option_exit();
+    }
+}
+
+sub manage_selection {
+    my ($self, %options) = @_;
+
+    my $output_error = "";
+    my $output = "";
+    my $items;
+
+    # VMs
+    if (defined($self->{option_results}->{vms})) {
+	$items = $options{custom}->azure_list_vms(
+	    resource_group => $self->{option_results}->{resource_group}
+	    );
+	$self->{vms} = {no_tag => 0, total => 0, name => ""};
+	foreach my $item (@{$items}) {
+	    $self->{vms}->{total}++;;
+	    next if (defined($self->{option_results}->{exclude_name}) && $self->{option_results}->{exclude_name} ne ''
+		     && $item->{name} =~ /$self->{option_results}->{exclude_name}/);
+	    next if (defined($item->{tags}->{finopsstartstop}));
```

Comment:
of course, line 69 is something i forgot while cleaning up
that should be:
next if (defined($item->{tags}->{$self->{option_results}->{tag_name}}));

Reply:
I wanted to be sure. Thank you for confirming this. 

Would allowing user to set a key/value pair could be relevant? 

Like `--tags='environment'` if I want to check the tag environment is set and `--tags='environment => dev'` if I want to check that the tag environment is present and that the value is dev?

Reply:
i guess it can be usefull in some occasions, but checking for the value of the tag should be optional.
it should be another argument, like --tag-value='dev' (that could be eventually a regexp).

Reply:
Hum, ok I see. I would prefer to keep a single option managing both. I'm afraid we would lost the ability to pair key and value if we set multiple --tag-name and --tag-value option (and the algo would be slower). 

Let me try a few things and I'll come back to you.

Reply:
Ok, I did it on my branch. 

It's possible to set the tags like this: 

--tags='Environment => DEV' --tags='Version' 

Basically check that Environment tag exists and that the value is dev, and check that the Version tag is also present, whatever its value. 

I will complete the hybrid benefit mode and then issue a PR to replace this one. 

Thanks again